ci/workflows/pull-request.yml

```diff
@@ -386,3 +386,16 @@ steps:
           config: ci/docker-compose.yml
           mount-buildkite-agent: true
     timeout_in_minutes: 20
+
+  # Sqlsmith differential testing
+  - label: "Sqlsmith Differential Testing"
+    command: "ci/scripts/sqlsmith-differential-test.sh -p ci-dev"
+    if: build.pull_request.labels includes "ci/run-sqlsmith-differential-tests"
+    depends_on:
+      - "build"
+    plugins:
+      - docker-compose#v4.9.0:
+          run: ci-flamegraph-env
+          config: ci/docker-compose.yml
+          mount-buildkite-agent: true
+    timeout_in_minutes: 40
```

Comment:
(bikeshedding) There seems to be many copy-paste in the CI yamls. Maybe we can use some scripts to generate them... 🤣

Reply:
Wait, why don't we create a new pipeline which can be triggered by both main and pr.